lenskit-core/src/main/java/org/grouplens/lenskit/data/source/TextDataSource.java

```diff
@@ -181,6 +182,7 @@ public static TextDataSource fromSpec(TextDataSourceSpec spec) {
         bld.setFormat(fmt);
         bld.setItemFile(spec.getItemFile());
         bld.setItemNameFile(spec.getItemNameFile());
+        bld.setHeaderLines(spec.getHeaderLines());
```

Comment:
This needs to be specified on the format, not on the builder. So it would be `fmt.setHeaderLines`. That will make the tests pass.

We could make an argument that `setFormat` should allow this to work. But that is a separate change, I think.